### PR TITLE
Path implementation

### PR DESCRIPTION
**Pull request by jwerle**

### CODE REVIEW

File: test/src/path.js [hunk 1]
```diff
@@ -11,5 +10,246 @@ test('path', (t) => {
...
+  )
+
+  t.equal(
+    path.win32.normalize('c:\\beep\\boop\\a\\b\\..\\..\\foo\\bar.js'),
+    'c:\\beep\\boop\\foo\\bar.js',
+    'relative Windows directory'
+  )
+
+  t.equal(
+    path.win32.normalize('c:\\beep\\boop\\a\\b\\..\\..\\foo\\bar\\'),
```
Comment (mribbons) [inline test/src/path.js:237]: I would say preserving the trailing slash is not required.
Including the trailing slash is a pre `path.join()` anti pattern.
We can make a decision here and say that a normalized path is one without a trailing slash.

File: test/src/process.js [hunk 2]
```diff
@@ -16,9 +16,6 @@ test('process.exit()', (t) => {
 })
 
 test('process.cwd', async (t) => {
```
Comment (mribbons) [inline test/src/process.js:18]: Can  we add a test to ensure `cwd` exists?

File: test/src/fs/promises.js [hunk 3]
```diff
@@ -65,8 +65,9 @@ if (process.platform !== 'ios') {
 
   test('fs.promises.readFile', async (t) => {
     const data = await fs.readFile(FIXTURES + 'file.txt')
+    const crlf = process.platform === 'win32' ? '\r\n' : '\n'
```
Comment (mribbons) [inline test/src/fs/promises.js:68]: `crlf` should be `eol` :)

File: test/src/path.js (hunk 1 above)
Comment (jwerle) [reply test/src/path.js:237]: We want to maintain parity with Node's `path` module here to keep up expectations. I also would think that if I give a trailing slash, that I get one back.

File: test/src/process.js (hunk 2 above)
Comment (jwerle) [reply test/src/process.js:18]: there are plenty tests for this already in `test/src/process.js`

File: test/src/fs/promises.js (hunk 3 above)
Comment (jwerle) [reply test/src/fs/promises.js]: 👍🏼

**Comment by jwerle** [comment]
Okay this one is good to go. Tests passing for all platforms!

**Review by mribbons — APPROVED** [review]

**Review by chicoxyzzy — APPROVED** [review]
:tada:

**Pull request merged**